serde_derive/src/internals/attr.rs

```diff
@@ -341,6 +356,43 @@ impl Container {
                             }
                         }
                     }
+                } else if meta.path == RENAME_ALL_FIELDS {
+                    // #[serde(rename_all_fields = "foo")]
+                    // #[serde(rename_all_fields(serialize = "foo", deserialize = "bar"))]
+                    match item.data {
+                        syn::Data::Enum(_) => {
+                            let one_name = meta.input.peek(Token![=]);
+                            let (ser, de) = get_renames(cx, RENAME_ALL, &meta)?;
+                            if let Some(ser) = ser {
+                                match RenameRule::from_str(&ser.value()) {
+                                    Ok(rename_rule) => {
+                                        rename_all_fields_ser_rule.set(&meta.path, rename_rule);
+                                    }
+                                    Err(err) => cx.error_spanned_by(ser, err),
+                                }
+                            }
+                            if let Some(de) = de {
+                                match RenameRule::from_str(&de.value()) {
+                                    Ok(rename_rule) => {
+                                        rename_all_fields_de_rule.set(&meta.path, rename_rule);
+                                    }
+                                    Err(err) => {
+                                        if !one_name {
+                                            cx.error_spanned_by(de, err);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        syn::Data::Struct(syn::DataStruct { struct_token, .. }) => {
+                            let msg = "#[serde(rename_all_fields)] can only be used on enums";
+                            cx.error_spanned_by(struct_token, msg);
+                        }
```

Comment:
Returning a success from `parse_nested_meta` closure in this case, without having parsed the rest of the `rename_all_fields` attribute, is going to cause bogus "expected ','" errors:

```console
error: #[serde(rename_all_fields)] can only be used on enums
 --> src/main.rs:5:1
  |
5 | struct Struct {}
  | ^^^^^^

error: expected `,`
 --> src/main.rs:4:27
  |
4 | #[serde(rename_all_fields = "asdf")]
  |                           ^
```

The whole attribute needs to be parsed before you validate it.